modular_skyrat/modules/colony_fabricator/code/machines/arc_furnace.dm

```diff
@@ -0,0 +1,193 @@
+	eject_contents()
+	return ..()
+
+/obj/machinery/arc_furnace/update_appearance()
+	. = ..()
+	cut_overlays()
+
+	if(length(contents))
+		var/image/overlayed_item = image(icon = contents[1].icon, icon_state = contents[1].icon_state)
+		overlayed_item.transform = matrix(, 0, 0, 0, 0.8, 0)
+		add_overlay(overlayed_item)
+
+	var/image/furnace_front_overlay = image(icon = icon, icon_state = "[operating ? "[base_icon_state]_overlay_active" : "[base_icon_state]_overlay"]")
+	add_overlay(furnace_front_overlay)
+
+/obj/machinery/arc_furnace/attackby(obj/item/attacking_item, mob/living/user, params)
+	if(operating)
+		balloon_alert(user, "furnace busy")
+		return TRUE
+
+	if(length(contents))
+		balloon_alert(user, "furnace full")
+		return TRUE
+
+	if(istype(attacking_item, /obj/item/stack/ore))
+		attacking_item.forceMove(src)
+		balloon_alert(user, "ore added")
+		update_appearance()
+		return TRUE
+
+	return ..()
+
+/obj/machinery/arc_furnace/ui_interact(mob/user)
+	. = ..()
+
+	if(operating || !user.can_perform_action(src, ALLOW_SILICON_REACH))
+		return
+	if(isAI(user) && (machine_stat & NOPOWER))
+		return
+
+	if(!length(contents))
+		balloon_alert(user, "it's empty!")
+		return
+
+	var/choice = show_radial_menu(user, src, radial_options, require_near = !issilicon(user))
+
+	// post choice verification
+	if(operating || !user.can_perform_action(src, ALLOW_SILICON_REACH))
+		return
+	if(isAI(user) && (machine_stat & NOPOWER))
+		return
+
+	usr.set_machine(src) // What does this even do??
+	switch(choice)
+		if("eject")
+			eject_contents()
+		if("use")
+			smelt_it_up(user)
+
+/obj/machinery/arc_furnace/proc/eject_contents()
+	if(operating)
+		return
+
+	playsound(loc, 'sound/machines/click.ogg', 15, TRUE, -3)
+
+	if(!length(contents))
+		return
+
+	var/atom/movable/thing_inside = contents[1]
+	thing_inside.forceMove(drop_location())
+	update_appearance()
+
+/obj/machinery/arc_furnace/proc/smelt_it_up(mob/user)
+	if(machine_stat & (NOPOWER|BROKEN))
+		balloon_alert(user, "button doesn't respond")
+		return
+	if(operating)
+		balloon_alert(user, "already smelting")
+		return
+
+	var/obj/item/stack/ore/ore_to_smelt = contents[1]
+	if(!istype(ore_to_smelt))
+		balloon_alert(user, "nothing to smelt")
+
+	operating = TRUE
+	/// How long the smelting is going to take based off the stack size
+	var/smelting_time = ore_to_smelt.amount * 1 SECONDS
+	loop(smelting_time)
+
+	soundloop.start()
+	set_light(l_range = 1.5)
+
+	update_appearance()
+
+/obj/machinery/arc_furnace/proc/loop(time)
+	if(machine_stat & (NOPOWER|BROKEN))
+		end_smelting()
+		return
+
+	if(!length(contents))
+		end_smelting()
+		return
+
+	if(time <= 0)
+		succeed_smelting()
+		return
+
+	time -= 1 SECONDS
+	use_power(active_power_usage)
+
+	var/turf/where_we_spawn_air = get_turf(src)
+	var/obj/item/stack/ore/ore_stack_to_check = contents[1]
+	switch(ore_stack_to_check.refined_type)
+		if(/obj/item/stack/sheet/mineral/silver)
+			where_we_spawn_air.atmos_spawn_air("n2=10;TEMP=1200")
```

Comment:
Okay that's super cool that it outputs hot air depending on certain minerals. Just make sure this is mentioned somewhere in-game so people don't think their machine is starting fires for no reason